controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -88,64 +95,67 @@ public SegmentHelper(final ConnectionFactory clientCF, HostControllerStore hostS
         return Controller.NodeUri.newBuilder().setEndpoint(host.getIpAddr()).setPort(host.getPort()).build();
     }
 
+    private void closeConnection(Reply reply, RawClient client) {
+        log.info("Closing connection as a result of receiving: {}", reply);
+        RawClient c;
+        synchronized (lock) {
+            c = client;
+            client = null;
+        }
+        if (c != null) {
+            try {
+                c.close();
+            } catch (Exception e) {
+                log.warn("Exception tearing down connection: ", e);
+            }
+        }
+    }
+
+    private boolean isReplyWanted(Reply reply, Class[] classArray) {
+        for (Class<?> c : classArray) {
+            if (c.isAssignableFrom(reply.getClass())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @SneakyThrows(ConnectionFailedException.class)
+    private Boolean transformReplyToBoolean(Reply reply, Class[] classArray, RawClient client) {
+        if (isReplyWanted(reply, classArray)) {
+            return true;
+        }
+
+        closeConnection(reply, client);
```

Comment:
It's not obvious from the method name that it will close the connection. It's even less obvious that it does so only if the reply is not wanted.

Reply:
i'm taking the following as the reference: `ConditionalOutputStreamImpl` and `SegmentMetadataClientImple`;

consider the original `SegmentHelper` implementation: if the reply is not wanted: the default way in replyProcessor is throw exception.

my think is: return false or throw exception - throw exception for getting more detailed logs. 

maybe change `transfromReplyToBoolean` => `transformReply` to avoid misleading caused by function name?